sdk/core/System.ClientModel/src/Convenience/KeyCredential.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+
+namespace System.ClientModel;
+
+public class KeyCredential
+{
+    private string _key;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KeyCredential"/> class.
+    /// </summary>
+    /// <param name="key">Key to use to authenticate with the Azure service.</param>
+    /// <exception cref="System.ArgumentNullException">
+    /// Thrown when the <paramref name="key"/> is null.
+    /// </exception>
+    /// <exception cref="System.ArgumentException">
+    /// Thrown when the <paramref name="key"/> is empty.
+    /// </exception>
+#pragma warning disable CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+    public KeyCredential(string key) => Update(key);
+#pragma warning restore CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
+
+    // Note: this is GetValue instead of GetKey to allow consistent naming
+    // across credential types.  For example, for NamedKeyCredential, we would have
+    // GetValues with two out params to enable atomicity in reading credential values.
+    public string GetValue() => Volatile.Read(ref _key);
+
+    public void Update(string key)
```

Comment:
@JoshLove-msft's comment got lost in the closed-PR but adding the TODO to change this to a Deconstruct method

Reply:
Update: I reviewed the concerns the BCL team had made about using a Deconstruct method.  Levi had specifically raised the concern that the Deconstruct pattern assumes a logical ordering and it wasn't clear what that ordering would be for some of the credential types.  Immo had responded that we could establish that in a constructor but acknowledged that people would need to read the docs which was a negative.

This seems like a mitigatable risk, so I will add the Deconstruct method instead of GetValue and if our architects approve it, we can discuss later with the BCL team if there is another preferred approach that would allow us to achieve consistency across our credential types.